src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java

```diff
@@ -0,0 +1,127 @@
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import jenkins.tasks.SimpleBuildStep;
+import org.apache.zeppelin.interpreter.InterpreterException;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+
+public class IPythonBuilder extends Builder implements SimpleBuildStep {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IPythonBuilder.class);
+
+    private final String code;
+    private final String filePath;
+    private final String parserType;
+
+    @DataBoundConstructor
+    public IPythonBuilder(String code,String filePath, String parserType) {
+        this.code = code;
+        this.filePath = filePath;
+        this.parserType = parserType;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public InterpreterManager getIPythonInterpreterManager(IPythonUserConfig jobUserConfig) throws InterpreterException {
+            InterpreterManager interpreterManager = new IPythonInterpreterManager(jobUserConfig);
+            interpreterManager.initiateInterpreter();
+            LOGGER.info("Connection initiated successfully");
+            return interpreterManager;
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener listener){
+        try {
+
+            // get the properties of the job
+            ServerJobProperty ipythonServerJobProperty = run.getParent().getProperty(ServerJobProperty.class);
+            String serverAddress = ipythonServerJobProperty.getServer().getServerAddress();
+            long launchTimeout = ipythonServerJobProperty.getServer().getLaunchTimeoutInMilliSeconds();
+            long maxResults = ipythonServerJobProperty.getServer().getMaxResults();
+            listener.getLogger().println(serverAddress);
+            // create configuration
+            IPythonUserConfig jobUserConfig = new IPythonUserConfig(serverAddress,launchTimeout,maxResults);
+            IPythonInterpreterManager interpreterManager = (IPythonInterpreterManager) getIPythonInterpreterManager(jobUserConfig);
+
+            listener.getLogger().println("Code Parser type : "+ parserType);
+            listener.getLogger().println("Executed code output : ");
+            if(parserType.equals("text")){
+                listener.getLogger().println(interpreterManager.invokeInterpreter(code));
+                interpreterManager.closeInterpreter();
+            }
+
+            //TODO for file parsers
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Symbol("python")
```

Comment:
This is supposed to be the unique identifier of the extension I think (hadn't used it before). There is (or there was?) a Python Plugin. Not sure if that plugin is using this identifier, but to avoid possible issues of name clashing, what about choosing another name?

The javadocs has a series of guidelines for choosing the right value, and how to format it.

![image](https://user-images.githubusercontent.com/304786/84836535-09866700-b08a-11ea-846a-d8dea1cf8f4b.png)

I found some examples on GitHub searching the jenkinsci org, and they were normally either related to the plug-in name, or to the class used :+1: